opengovsg/postmangovsg
# Verify backup database dump in Google Cloud

## Problem

The backup db dumps in google cloud storage need to be verified in order to prevent silent failures.

Closes #705

## Solution

- Get gcloud backup account service key from Secret Manager 
- Downloads the latest backup from backup GCS bucket using key
- Use KMS private key to decrypt the dump from GCS bucket
- Run pg_restore on the dump and check that it completes successfully
- Run functional checks against the restored database (API calls, simple queries, etc)


**Refer to [Setup GCP services for verifying backups #758](https://github.com/opengovsg/postmangovsg/issues/758) for more details on how to run the scripts and setup services on GCP.**

## Testing
- a service account key is required to test locally
- build and run the docker image
